pkg/sharding/sharding.go

```diff
@@ -19,71 +19,81 @@ import (
 // the TreeID and the UUID. The TreeID is a hex-encoded uint64 (8 bytes)
 // referring to the specific trillian tree (also known as log or shard) where
 // the artifact can be found. The UUID is a hex-encoded 32-byte number
 // referring to the artifact's merkle leaf hash from trillian. Artifact lookup
 // by UUID occurs by finding the UUID within the tree specified by the TreeID.
 //
-// A FullID is 40 bytes long and looks like this:
+// An EntryID is 40 bytes long and looks like this:
 // FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF FFFF
 // |_______  ________| |_____________________________________  ______________________________________|
 //         \/                                                \/
 // TreeID (8 bytes, hex)                             UUID (32 bytes, hex)
 
 const TreeIDHexStringLen = 16
 const UUIDHexStringLen = 64
-const FullIDHexStringLen = TreeIDHexStringLen + UUIDHexStringLen
+const EntryIDHexStringLen = TreeIDHexStringLen + UUIDHexStringLen
 
 // TODO: replace this with the actual LogRanges struct when logic is hooked up
-var dummy = api.LogRanges{
-	Ranges: []api.LogRange{},
+var dummy = LogRanges{
+	Ranges: []LogRange{},
 }
 
-type FullID struct {
+type EntryID struct {
 	TreeID string
 	UUID   string
 }
 
-func CreateFullID(treeid string, uuid string) (FullID, error) {
+func CreateEntryID(treeid string, uuid string) (EntryID, error) {
 	if len(treeid) != TreeIDHexStringLen {
 		err := fmt.Errorf("invalid treeid len: %v", len(treeid))
-		return createEmptyFullID(), err
+		return createEmptyEntryID(), err
 	}
 
 	if len(uuid) != UUIDHexStringLen {
 		err := fmt.Errorf("invalid uuid len: %v", len(uuid))
-		return createEmptyFullID(), err
+		return createEmptyEntryID(), err
 	}
 
 	if _, err := hex.DecodeString(treeid); err != nil {
 		err := fmt.Errorf("treeid is not a valid hex string: %v", treeid)
-		return createEmptyFullID(), err
+		return createEmptyEntryID(), err
 	}
 
 	if _, err := hex.DecodeString(uuid); err != nil {
 		err := fmt.Errorf("uuid is not a valid hex string: %v", uuid)
-		return createEmptyFullID(), err
+		return createEmptyEntryID(), err
 	}
 
-	return FullID{
+	return EntryID{
 		TreeID: treeid,
 		UUID:   uuid}, nil
 }
 
-func createEmptyFullID() FullID {
-	return FullID{
+func createEmptyEntryID() EntryID {
+	return EntryID{
 		TreeID: "",
 		UUID:   ""}
 }
 
-func PrependActiveTreeID(uuid string) (FullID, error) {
+func PrependActiveTreeID(uuid string) (EntryID, error) {
 	// TODO: Update this to be the global LogRanges struct
 	active := dummy.ActiveIndex()
-	return CreateFullID(strconv.FormatUint(active, 10), uuid)
+	return CreateEntryID(strconv.FormatUint(active, 10), uuid)
+}
+
+func (e EntryID) ReturnEntryIDString() string {
+	return e.TreeID + e.UUID
+}
+
+// Returns UUID (with no prepended TreeID) from a UUID or EntryID string
+func GetUUIDFromIDString(id string) (string, error) {
```

Comment:
Yeah, this is a good idea. I'm just not sure yet how to do it without tearing out the entire `pflags.go` logic and moving it into sharding. `validateID` relies on `validateString`, which uses the `validationFunc`, `baseValue`, etc. and I can't import those into `sharding` because `rekor-cli` has `sharding` as a dependency. Writing a duplicate version of `validateString` in `sharding` also seems suboptimal. Possibly would make sense to move `pflags.go` into a package accessible to both `sharding` and `rekor-cli`? What do you think?

Reply:
I guess this function could also just move into `pflags`.

Reply:
I think in order to do this, `pflags.go` and related files would have to move "up" and out of `rekor-cli` into a new `pkg/validation` or `cmd/validation` because that way `GetUUIDFromIDString` could be used in both `cmd/rekor-cli/app/get.go` and `pkg/api/entries.go` without import cycles. Still experimenting with this.

Reply:
I ended up using `hex.DecodeString` instead of moving the `pflags` logic around.